src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -39,90 +39,216 @@
+                List<ArchivedRecordingInfo> temp =
+                        files.stream()
+                                .map(
+                                        file -> {
+                                            try {
+                                                return new ArchivedRecordingInfo(
+                                                        subdirectory,
+                                                        file,
+                                                        webServer.getArchivedReportURL(file),
+                                                        webServer.getArchivedDownloadURL(file));
+                                            } catch (SocketException
+                                                    | UnknownHostException
+                                                    | URISyntaxException e) {
+                                                logger.warn(e);
+                                                return null;
+                                            }
+                                        })
+                                .filter(Objects::nonNull)
+                                .collect(Collectors.toList());
+                archivedRecordings.addAll(temp);
+            }
+            future.complete(archivedRecordings);
+        } catch (ArchivePathException e) {
+            future.completeExceptionally(e);
+        }
 
-        return saveName;
+        return future;
     }
 
-    public void deleteRecording(ConnectionDescriptor connectionDescriptor, String recordingName)
-            throws Exception {
+    public Future<Path> getRecordingPath(String recordingName) throws Exception {
+        CompletableFuture<Path> future = new CompletableFuture<>();
 
-        targetConnectionManager.executeConnectedTask(
-                connectionDescriptor,
-                connection -> {
-                    Optional<IRecordingDescriptor> descriptor =
-                            this.getDescriptorByName(connection, recordingName);
+        try {
+            Path archivedRecording = null;
+            Boolean recordingFound = false;
+            List<String> subdirectories = this.fs.listDirectoryChildren(recordingsPath);
+            for (String subdirectory : subdirectories) {
+                List<String> files =
+                        this.fs.listDirectoryChildren(recordingsPath.resolve(subdirectory));
 
-                    if (descriptor.isPresent()) {
-                        connection.getService().close(descriptor.get());
-                        reportService.delete(connectionDescriptor, recordingName);
-                    } else {
-                        throw new RecordingNotFoundException(recordingName);
+                for (String file : files) {
+                    if (recordingName.equals(file)) {
+                        archivedRecording = recordingsPath.resolve(subdirectory + "/" + file);
```

Comment:
This part doesn't look right: `recordingsPath.resolve(subdirectory + "/" + file)`

Would this be equivalent to `recordingsPath.resolve(subdirectory).resolve(file)`?